lib/services/project-changes-service.ts

```diff
@@ -227,10 +228,15 @@ export class ProjectChangesService implements IProjectChangesService {
 
 			let fileStats = this.$fs.getFsStats(filePath);
 
-			let changed = fileStats.mtime.getTime() >= this._outputProjectMtime || fileStats.ctime.getTime() >= this._outputProjectCTime;
-			if (!changed) {
-				let lFileStats = this.$fs.getLsStats(filePath);
-				changed = lFileStats.mtime.getTime() >= this._outputProjectMtime || lFileStats.ctime.getTime() >= this._outputProjectCTime;
+			let changed =  fileStats.mtime.getTime() > this._outputProjectMtime ||
+				fileStats.ctime.getTime() >= this._outputProjectCTime;
+
+			if (this.$hostInfo.isDarwin && !fileStats.isDirectory() && !changed) {
+				const parentDirPath: string = path.dirname(filePath);
+				const parentDirStat = this.$fs.getFsStats(parentDirPath);
```

Comment:
@rosen-vladimirov I did this as a temporary solution because I needed to test the fix on Mac and I needed to pull the code from it. I'm sorry if I inconvenienced you. I'm currently working on an optimized, permanent solution and will ping you when ready.